src/libraries/System.Private.CoreLib/src/System/Buffers/Text/Utf8Parser/Utf8Parser.Integer.Signed.D.cs

```diff
@@ -192,146 +201,132 @@ private static bool TryParseInt16D(ReadOnlySpan<byte> source, out short value, o
             return true;
         }
 
-        private static bool TryParseInt32D(ReadOnlySpan<byte> source, out int value, out int bytesConsumed)
+        private static bool TryParseInt32D(in ReadOnlySpan<byte> refToSource, out int value, out int bytesConsumed)
```

Comment:
I suspect that the `in` is deoptimization everywhere else, except Windows x64.

Reply:
In other words, passing ReadOnlySpan as `in` does not sounds like a good idea.

Reply:
I agree this is not a good optimization, and we should back it out if we take this PR. But it's hard to argue with the data that shows minimizing parameter shuffling (see assembly in https://github.com/dotnet/runtime/pull/32843#issuecomment-591656875) has a measurable impact on performance.